dotnet/src/Agents/Abstractions/AgentChat.cs

```diff
@@ -134,47 +186,73 @@ await foreach (var message in channel.InvokeAsync(agent, cancellationToken).Conf
         }
         finally
         {
-            Interlocked.Exchange(ref this._isActive, 0);
+            this.ClearActivitySignal();
         }
-    }
-
-    private async Task<AgentChannel> GetChannelAsync(Agent agent, CancellationToken cancellationToken)
-    {
-        var channelKey = this.GetAgentHash(agent);
 
-        if (this._agentChannels.TryGetValue(channelKey, out var channel))
+        async Task<AgentChannel> GetOrCreateChannelAsync()
         {
-            await this._broadcastQueue.EnsureSynchronizedAsync(new ChannelReference(channel, channelKey)).ConfigureAwait(false);
-        }
-        else
-        {
-            channel = await agent.CreateChannelAsync(cancellationToken).ConfigureAwait(false);
-
-            if (this.History.Count > 0)
+            string channelKey = this.GetAgentHash(agent);
+            AgentChannel channel = await this.SynchronizeChannelAsync(channelKey, cancellationToken).ConfigureAwait(false);
+            if (channel == null)
             {
-                await channel.ReceiveAsync(this.History, cancellationToken).ConfigureAwait(false);
+                channel = await agent.CreateChannelAsync(cancellationToken).ConfigureAwait(false);
+                this._agentChannels.Add(channelKey, channel);
+
+                if (this.History.Count > 0)
+                {
+                    await channel.ReceiveAsync(this.History, cancellationToken).ConfigureAwait(false);
+                }
             }
 
-            this._agentChannels.Add(channelKey, channel);
+            return channel;
         }
+    }
 
-        return channel;
+    /// <summary>
+    /// Clear activity marker.
+    /// </summary>
+    private void ClearActivitySignal()
+    {
+        Interlocked.Exchange(ref this._isActive, 0);
     }
 
-    private string GetAgentHash(Agent agent)
+    /// <summary>
+    /// Test to ensure chat is not concurrently active and throw exception if it is.
+    /// If not, activity is signaled.
```

Comment:
This is a non obvious concept, can you expand the comment?

Reply:
Done